Document how to migrate passwords from Synapse with a `pepper` set (#4353)
element-hq/matrix-authentication-service: docs/setup/migration.md

Synapse uses bcrypt as its password hashing scheme while MAS defaults to using the newer argon2id.
You will have to configure the version 1 scheme as bcrypt for migrated passwords to work.
It is also recommended that you keep argon2id as version 2 so that once users log in, their hashes will be updated to the newer recommended scheme.

> If you have set a pepper in the Synapses password_config section of your homeserver.yaml then you need to specify this pepper as the secret field for your bcrypt scheme. Otherwise logins with the correct Password will fail.

Reviewer: ```suggestion
If you have a `pepper` set in the `password_config` section of your Synapse config, then you need to specify this `pepper` as the `secret` field for your `bcrypt` scheme.
```

  schemes:
  - version: 1
    algorithm: bcrypt

>     # Optional, The secret field is the equivalent to Synapses password_config pepper.
>     secret: secretPepperValue

Reviewer: ```suggestion
    # Optional, must match the `password_config.pepper` in the Synapse config
    #secret: secretPepperValue
```
